src/metrics/include/metrics/metrics.hpp

```diff
@@ -34,6 +34,9 @@ namespace genny::metrics {
  */
 namespace v1 {
 
+template <typename Clocksource>
```

Comment:
A forward declaration was necessary since metrics.hpp and operation.hpp import each other.